wandb/sdk/launch/agent/agent.py

```diff
@@ -53,17 +79,17 @@ def __init__(self, api: Api, config: Dict[str, Any]):
         self._project = config.get("project")
         self._api = api
         self._base_url = self._api.settings().get("base_url")
+        # _jobs includes scheduler jobs as well
```

Comment:
Wouldn't it be simpler to manage if they were separate?

Reply:
Should we launch schedulers anytime we are under the scheduler cap, or should we only launch scheduler jobs when there is room under our job cap? Its pretty obvious the answer is the former, but this does complicate our `pop_from_queue` logic, as we must first inspect a job to determine whether it is a scheduler or not. For example, if our max-jobs is set to 2 and max-schedulers is set to 1:

launch agent picks up and runs job 1
launch agent picks up and runs job 2

now, the launch agent still has room in the cap to try to find a scheduler, but must pop and then NOT ack a ton of jobs to find the next scheduler.

However, if we say that schedulers can only be run if there is room in BOTH caps, we avoid this issue entirely. Why this might make sense is that even with a scheduler running, the agent will do **nothing** unless there is room in the job cap to start injesting the jobs created by the scheduler. Obviously if there are multiple agents this gets even more confusing.... @KyleGoyette

Reply:
Even so, can't we perform that check on both separate dicts/counters before popping a run queue item?  Otherwise, I agree. Please add a brief comment explaining this decision.

Reply:
Well we always know if we don't want to pop at all (if both are at cap). The confusing bit is when we have room in the scheduler cap, but no room in the job cap. What should we do then?